src/SqlPersistence.Tests/CharArrayTextWriterPerformanceTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NServiceBus;
+using NServiceBus.Persistence.Sql;
+using NServiceBus.Sagas;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+
+[Explicit("This is a long running performance test")]
+public class CharArrayTextWriterPerformanceTests
+{
+    [TestCaseSource(nameof(Dialects))]
+    public void PerfTest(SqlDialect dialect)
+    {
+        var sagaInfo = BuildSagaInfo<S,S.SagaData>(dialect);
+        var data = new S.SagaData
+        {
+            StringProperty = "Some bigger test",
+            GuidProperty = Guid.NewGuid(),
+            OriginalMessageId = Guid.NewGuid().ToString(),
+            Originator = "originator",
+            IntProperty = 5,
+            Id = Guid.NewGuid()
+        };
+
+        // warm up
+        using (var cmd = new CommandWrapper(null, dialect))
+        {
+            Console.WriteLine(dialect.BuildSagaData(cmd, sagaInfo, data));
+        }
+
+        var sw = Stopwatch.StartNew();
+        for (var i = 0; i < 10_000_000; i++)
+        {
+            using (var cmd = new CommandWrapper(null, dialect))
+            {
+                dialect.BuildSagaData(cmd, sagaInfo, data);
+            }
+        }
+        Console.WriteLine($"Test took: '{sw.Elapsed}'");
+    }
+
+    public class S :
+        SqlSaga<S.SagaData>,
+        IAmStartedByMessages<AMessage>
+    {
+        public class SagaData : ContainSagaData
+        {
+            public string StringProperty { get; set; }
+            public int IntProperty { get; set; }
+            public Guid GuidProperty { get; set; }
+        }
+
+        protected override string CorrelationPropertyName => nameof(SagaData.StringProperty);
+
+        protected override string TableSuffix => "TableSuffix";
+
+        protected override void ConfigureMapping(IMessagePropertyMapper mapper)
+        {
+            mapper.ConfigureMapping<AMessage>(_ => _.StringId);
+        }
+
+        public Task Handle(AMessage message, IMessageHandlerContext context)
+        {
+            return Task.FromResult(0);
+        }
+    }
+
+    public class AMessage : IMessage
+    {
+        public string StringId { get; set; }
+    }
+
+    public static IEnumerable<ITestCaseData> Dialects()
```

Comment:
Add Oracle.

Reply:
I picked up MySql only as a comparison. Do you think it's worth to add Oracle?

Reply:
OK, now I understand the driver specific aspect. We can stay with MySQL only.